File: objc/objcsearch/Tests/objcsearchTests/SearchFileTests.m
#import <XCTest/XCTest.h>
#import "FileTypes.h"
#import "SearchFile.h"

@interface SearchFileTests : XCTestCase
//@property FileTypes *fileTypes;
@end

@implementation SearchFileTests

- (void)testSearchFileAbsPath {
    NSString *path = @"/Users/cary/src/xsearch/objc/objcsearch/objcsearch/SearchFile.m";
    SearchFile *searchFile = [[SearchFile alloc]
                              initWithFilePath:path
                              fileType:FileTypeCode];
    NSString *searchFileString = [searchFile description];
    XCTAssert([searchFileString isEqualToString:path]);
}

- (void)testSearchFileTildePath {
    NSString *path = @"~/src/xsearch/objc/objcsearch/objcsearch/SearchFile.m";
    SearchFile *searchFile = [[SearchFile alloc]
                              initWithFilePath:path
                              fileType:FileTypeCode];
    NSString *searchFileString = [searchFile description];
    XCTAssert([searchFileString isEqualToString:path]);
}

- (void)testSearchFileRelPath1 {
    NSString *path = @"./SearchFile.m";
    SearchFile *searchFile = [[SearchFile alloc]
                              initWithFilePath:path
                              fileType:FileTypeCode];
    NSString *searchFileString = [searchFile description];
    XCTAssert([searchFileString isEqualToString:path]);
}

- (void)testSearchFileRelPath2 {
    NSString *path = @"../SearchFile.m";
    SearchFile *searchFile = [[SearchFile alloc]
                              initWithFilePath:path
                              fileType:FileTypeCode];
    NSString *searchFileString = [searchFile description];
    XCTAssert([searchFileString isEqualToString:path]);
}

@end

File: swift/swiftsearch/Tests/swiftsearchTests/SearchFileTests.swift
//
//  swiftsearchTests.swift
//  swiftsearchTests
//
//  Created by Cary Clark on 5/18/15.
//  Copyright (c) 2015 Cary Clark. All rights reserved.
//

import Cocoa
import XCTest

import swiftsearch

class SearchFileTests: XCTestCase {

    func testSearchFileAbsPath() {
        let path = "/Users/cary/src/xsearch/swift/swiftsearch/Sources/swiftsearch/SearchFile.swift"
        let searchFile = SearchFile(filePath: path, fileType: FileType.code)
        XCTAssertEqual(path, searchFile.description())
    }

    func testSearchFileTildePath() {
        let path = "~/src/xsearch/swift/swiftsearch/Sources/swiftsearch/SearchFile.swift"
        let searchFile = SearchFile(filePath: path, fileType: FileType.code)
        XCTAssertEqual(path, searchFile.description())
    }

    func testSearchFileRelPath1() {
        let path = "./SearchFile.swift"
        let searchFile = SearchFile(filePath: path, fileType: FileType.code)
        XCTAssertEqual(path, searchFile.description())
    }

    func testSearchFileRelPath2() {
        let path = "../SearchFile.swift"
        let searchFile = SearchFile(filePath: path, fileType: FileType.code)
        XCTAssertEqual(path, searchFile.description())
    }
}
